tests/Simpleue/Mocks/JobSpy.php

```diff
@@ -11,12 +11,18 @@
 class JobSpy implements Job {
 
     private $manageCounter;
+    private $quitCount;
+    private $testSignal;
 
     public function _construct() {
         $this->manageCounter = 0;
     }
 
     public function manage($job) {
+        if ($this->quitCount && (($this->manageCounter+1) === $this->quitCount)) {
```

Comment:
The idea was to send the kill signal before the counter was incremented so that we can ensure the counter gets incremented anyways because we handled it. Thus the placement of if.